### PR TITLE
Remove security parameter

### PR DESCRIPTION
**Pull request by alxdavids**
- Remove mention of security parameter (addresses #84, #87)
- Document setup procedure
- Remove protocol overview (superfluous and notation was inconsistent)
- Shorten ciphersuite spec

cc @armfazh as initial reporter

**Should be merged into #91 before that change is merged**

### CODE REVIEW

File: draft-irtf-cfrg-voprf.md [hunk 1]
```diff
@@ -552,35 +551,53 @@ Steps enclosed in `[[ ]]` are REQUIRED for verifiability.
...
+any Client that connects to it.
+
+The Server SHOULD sample a new PrivateKey object corresponding to its
+secret input. This value can be used across multiple instantiations of
+the protocol. If the server ciphersuite supports verifiability, then it
+interprets its private key as a Scalar `k` in `GF(p)` and computes:
+
+~~~
+PK = k * GG.Generator()
+public_key = GG.Serialize(PK)
```
Comment (chris-wood) [inline draft-irtf-cfrg-voprf.md:590]: Why `public_key` for the serialized version and `PK` for the deserialized one?

File: draft-irtf-cfrg-voprf.md [hunk 2]
```diff
@@ -552,35 +551,53 @@ Steps enclosed in `[[ ]]` are REQUIRED for verifiability.
...
+~~~
 
-We restate the (V)OPRF protocol using a concrete API that we describe in
-{{algorithms}} below.
+The Server MUST publish `public_key` so that any valid Client can use it
+during the protocol to verify messages from Server.
```
Comment (chris-wood) [inline draft-irtf-cfrg-voprf.md:594]: This seems like a weird MUST. What if there's an application that genuinely does not require verifiability? I think we should reframe this a bit, e.g., "Servers that support verifiability MUST make `public_key` available to clients." (And I'd drop the serialization requirement at the same time.)

File: draft-irtf-cfrg-voprf.md [hunk 3]
```diff
@@ -780,39 +802,45 @@ Output:
...
-VOPRF protocols using additive blinding rather than multiplicative
-blinding. In fact, the only algorithms that need to change are Blind and
-Unblind (and similarly for the VOPRF variants).
-
-We define the additive variants of the algorithms in {{algorithms}}
-below, along with a new algorithm Preprocess. The Proprocess algorithm
-can take place offline and before the rest of the OPRF protocol. The
-Blind algorithm takes the proprocessed values as inputs, but the
-signature of Unblind remains the same.
+Let `H` refer to the function `GG.HashToGroup`, in {{pog}} we assume
```
Comment (chris-wood) [inline draft-irtf-cfrg-voprf.md:812]: This section seems to revert to the old scalar multiplication notation. Is that an intentional regression?

File: draft-irtf-cfrg-voprf.md [hunk 4]
```diff
@@ -1118,17 +1145,17 @@ of an OPRF.
...
 need to be made.
 
 ### Computational hardness assumptions {#assumptions}
 
 Each assumption states that the problems specified below are
-computationally difficult to solve in relation to sp (the security
-parameter). In other words, the probability that an adversary has in
-solving the problem is bounded by a function negl(sp), where negl(sp) <
-1/f(sp) for all polynomial functions f().
+computationally difficult to solve in relation to a paritcular choice of
```
Comment (chris-wood) [inline draft-irtf-cfrg-voprf.md:1155]:
```suggestion
computationally difficult to solve in relation to a particular choice of
```

File: draft-irtf-cfrg-voprf.md [hunk 5]
```diff
@@ -1118,17 +1145,17 @@ of an OPRF.
...
+security parameter `sp`. In other words, the probability that an
+adversary has in solving the problem is bounded by a function negl(sp),
+where negl(sp) < 1/f(sp) for all polynomial functions f().
```
Comment (chris-wood) [inline draft-irtf-cfrg-voprf.md:1158]:
```suggestion
security parameter `sp`.
```

File: draft-irtf-cfrg-voprf.md (hunk 3 above)
Comment (alxdavids) [reply draft-irtf-cfrg-voprf.md:812]: Will fix

File: draft-irtf-cfrg-voprf.md [hunk 6]
```diff
@@ -503,89 +502,69 @@ multiplication into all elliptic curve operations.
...
+interprets its private key as a Scalar `k` in `GF(p)` and computes:
 
-# Protocol API {#api}
+~~~
+PK = k * GG.Generator()
+~~~
 
-We restate the (V)OPRF protocol using a concrete API that we describe in
-{{algorithms}} below.
+Servers that support verifiability MUST make public_key available to
```
Comment (chris-wood) [inline draft-irtf-cfrg-voprf.md:560]:
```suggestion
Servers that support verifiability MUST make `PK` available to
```

File: draft-irtf-cfrg-voprf.md [hunk 7]
```diff
@@ -911,8 +901,14 @@ Steps:
  4. Output unblinded_tokens
 ~~~
 
-Notice that Unblind computes (Z-PKr) = k(H_1(x)+r * G) - rk * G =
-kH_1(x) by the commutativity of scalar multiplication in GG. 
+Let `P = GG.HashToGroup(x)`. Notice that Unblind computes:
+
+~~~
+(Z-PKR) = k(P + r * G) - (rk) * G = k * P
```
Comment (chris-wood) [inline draft-irtf-cfrg-voprf.md:907]:
```suggestion
Z - PKR = k(P + r * G) - (rk) * G = k * P
```

**Pull request merged**